package/MDAnalysis/core/groups.py

```diff
@@ -349,7 +349,15 @@ class _ImmutableBase(object):
     #  cache lookup if the class is reused (as in ag._derived_class(...)).
     __new__ = object.__new__
 
-
+def check_pbc_and_unwrap(function):
```

Comment:
Note that `pbc` and `unwrap` are only mutually exclusive if `compound='group'`.
There is nothing wrong with having both `pbc` and `unwrap` set to `True` if `compound` is e.g. `molecules`.

Reply:
How would this work?

Reply:
just as the docs say: "[...] Instead, the resulting position vectors will be moved to the primary unit cell after calculation."
In other words: 
1. `unwrap=True` => Unwrap components.
2. Compute center of each component
3. `pbc=True` => Wrap the computed centers into the primary unit cell.

Reply:
We're renaming `pbc=True` to `wrap=True`, having `method(wrap=True, unwrap=True)` just makes no sense to read.  Doing this would have to be two function calls...

```python
ag.unwrap()  # make molecules whole
ag.wrap(compound='group')  # move whole molecules into primary unit cell
```
```